src/pset/mod.rs

```diff
@@ -733,7 +733,7 @@ mod tests {
         // Initially secp context and rng global state
         let secp = secp256k1_zkp::Secp256k1::new();
         #[allow(deprecated)]
-        let mut rng = rand::ChaChaRng::seed_from_u64(0);
+        let mut rng = rand::rngs::StdRng::seed_from_u64(0);
```

Comment:
according to docs StdRng use ChaCha, I think it's equivalent

Reply:
I've errors in examples in comparison against test vectors ( `pset_blind_coinjoin.rs` and `raw_blind.rs`) which I think it's because those are not equivalent... 

```
  left: `Global { tx_data: TxData { version: 2, fallback_locktime: None, input_count: 2, output_count: 5, tx_modifiable: None }, version: 2, xpub: {}, scalars: [Tweak(a9fe9d44d70fd0efea5c2313c7235b02475b81bd243d2004b853821cee0b2047)], elements_tx_modifiable_flag: None, proprietary: {}, unknown: {} }`,
 right: `Global { tx_data: TxData { version: 2, fallback_locktime: None, input_count: 2, output_count: 5, tx_modifiable: None }, version: 2, xpub: {}, scalars: [Tweak(0f56f3a1f798b585d605591085469b0e3d0c3f46c3bdf230773056302acccdb8)], elements_tx_modifiable_flag: None, proprietary: {}, unknown: {} }`', examples/pset_blind_coinjoin.rs:274:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I can't keep rand 0.6 because it would break against secp256k1_zkp which is now 0.8

Any suggestion?

Reply:
I can look into finding an equivalent here. 0.8 rand should have some equivalent PRNG.

Reply:
There is a deprecation notice on rand 0.6 referring to using the `rand_chacha` create, I tested it and it looks compatible:
```
  let mut rng = rand_chacha::ChaChaRng::seed_from_u64(0);
    assert_eq!(rng.next_u32(), 2180380594);

    let mut rng = rand::ChaChaRng::seed_from_u64(0);
    assert_eq!(rng.gen::<u32>(), 2180380594);

    let mut rng = rand::rngs::StdRng::seed_from_u64(0);
    assert_eq!(rng.gen::<u32>(),3384286946);
```
 
 So I added c187bbb9c6b6c5167562d316e46af23222c49b8f
 
 and `raw_blind.rs` example works, however, `pset_blind_coinjoin.rs` still doesn't!

Reply:
I'm tempted to add a commit which removes the `rand_chacha` dep, sticks in some custom crappy RNG, and updates the test vectors. We'd still have a commit that used `rand_chacha` so that readers could check that the change in tests was only due to the change in RNG, rather than anything else.

Reply:
Maybe that should just be a separate PR..

Reply:
https://github.com/ElementsProject/rust-elements/pull/148

Reply:
> pset_blind_coinjoin.rs still doesn't!

FYI this was due to wrongdoing in the "fix warning" commit